autofill/autofill-impl/src/main/java/com/duckduckgo/autofill/impl/ui/credential/management/viewing/AutofillManagementCredentialsMode.kt

```diff
@@ -354,26 +363,47 @@ class AutofillManagementCredentialsMode : DuckDuckGoFragment(R.layout.fragment_a
         invalidateMenu()
     }
 
+    private suspend fun preloadFavicon(credentials: LoginCredentials) {
+        withContext(dispatchers.io()) {
+            val size = resources.getDimensionPixelSize(dimen.toolbarIconSize)
+            val placeholder = generateDefaultFavicon(credentials, size)
+            withContext(dispatchers.main()) {
+                getActionBar()?.setLogo(BitmapDrawable(resources, placeholder))
```

Comment:
`BitmapDrawable(resources, placeholder)`--- Maybe this should be done outside of the main thread? It should only be the `actionBar()?.setLogo()` call that requires `main`.

e.g., 

```
private suspend fun preloadFavicon(credentials: LoginCredentials) {
        withContext(dispatchers.io()) {
            val size = resources.getDimensionPixelSize(dimen.toolbarIconSize)
            val placeholder = generateDefaultFavicon(credentials, size)
            val bitmapDrawable = BitmapDrawable(resources, placeholder)
            withContext(dispatchers.main()) {
                getActionBar()?.setLogo(bitmapDrawable)
            }
        }
    }
```